store/evm_aux/store.go

```diff
@@ -49,51 +71,51 @@ type ChildTxRef struct {
 }
 
 type EvmAuxStore struct {
-	db *leveldb.DB
+	db          dbm.DB
+	store       *atomicKVStore
+	maxReceipts uint64
 }
 
-func NewEvmAuxStore(db *leveldb.DB) *EvmAuxStore {
-	return &EvmAuxStore{db: db}
+func NewEvmAuxStore(db dbm.DB, maxReceipts uint64) *EvmAuxStore {
+	return &EvmAuxStore{
+		db:          db,
+		store:       NewAtomicKVStore(db),
```

Comment:
Or you just write everything in `CommitReceipts` to a single batch, and we avoid introducing yet another store.

Reply:
I was trying to replace `transaction` with `batch` but the logic to limit the number of receipts need to have access to the newly added receipts to calculate head and tail of linked list before  committing the result, transaction or cache is needed.

Reply:
Huh? Everything happens once per block in `CommitReceipts`, why can't all the linked list computation happen in there?

Reply:
Really don't want to touch the old linked list logic but yeah I should get it done the right way.